### PR TITLE
Support keyword assignment

### PR DESCRIPTION
**Pull request by jonathantneal**
contentKeyword, default 'mixin-content'
defineKeyword, default ‘define-mixin’
usageKeyword, default 'mixin'

### CODE REVIEW

**Comment by ai** [comment]
What is user case for this changes?

**Comment by jonathantneal** [comment]
I am using this plugin in [precss](https://github.com/jonathantneal/precss), but I would like parity with the Sass syntax.

```js
{
	defineKeyword: 'mixin',
	usageKeyword: 'include'
}
```

```scss
@mixin black {
	color: black;
}

a {
	@include black;
}
```

This allows parity without disturbing any existing users.

**Comment by ai** [comment]
There is important reason, why I select this keywords.

In Sass we control all stack, so we definitely know, that nobody will use `@include` keyword.

In PostCSS we can not think so. We has many independent plugins with different custom at-rules. This is why all postcss-mixins keywords has mixin word inside.

I understand that Sass compatible keyword will be easy to learn. But it creates very important long term problems of name conflicts.

Of course, we can make exclusion for Sass keywords, but I do not see a point, because anyway we will not make one-to-one compatibility.


Maybe we can solve this problem in different way, so I am open for suggestions. But we need to fix name conflicts problem first.

Why you think it is important to have same keywords?

**Comment by jonathantneal** [comment]
I don’t have much more to add. For my plugin, using the Sass keyword will make it easier to learn, but I’m making it configurable so not everyone has to follow my conventions or your conventions. For my purposes, it doesn’t help people to learn another keyword for doing essentially the same thing. That’s all.

**Comment by ai** [comment]
@jonathantneal I think option is a bad way. If we need Sass keeywords, we can just change plugin to support them. For example, it can works with `@content` and `@mixin-content`.

But we still has compatibility issue. Any ideas how to fix it?

File: index.js [hunk 1]
```diff
@@ -155,6 +155,18 @@ module.exports = postcss.plugin('postcss-mixins', function (opts) {
...
+    }
+
+    if ( !( 'defineKeyword' in opts ) ) {
+        opts.defineKeyword = 'define-mixin';
+    }
+
+    if ( !( 'usageKeyword' in opts ) ) {
+        opts.usageKeyword = 'mixin';
+    }
+
```
Comment (ben-eb) [inline index.js]: Suggestion to use https://github.com/sindresorhus/object-assign to handle the defaults, instead:

```js
var assign = require('object-assign');
// ...
opts = assign({
  contentKeyword: 'mixin-content',
  defineKeyword: 'define-mixin',
  usageKeyword: 'mixin'
}, opts);
```

**Comment by jonathantneal** [comment]
The `@mixin` keyword has a different meaning in Sass than it does in PostCSS, so an option will be necessary.

Could the option be a switch?

```js
{
  syntax: "postcss" // alternatively "sass"?
}
```

**Comment by ai** [comment]
Ouh, we can’t support both, because `@mixin` has different meaning

**Pull request closed without merging**